### ISSUE TITLE
onBecome(Un)Observed didn't trigger when use number as key of observable map

### ISSUE THREAD

**Issue by gespiton** [mobxjs/mobx#2067]
[x] Issue:
  * [x] a **minimal** sample reproduction. **[sandbox](https://codesandbox.io/s/mobx-playground-110jr)**
  * [x] Elaborate on your issue. What behavior did you expect?

when use number as key of observable map, onBecomeObserved didn't trigger callback, but it worked fine when using string as key 
  * [x] State the versions of MobX and relevant libraries. Which browser / node / ... version?

mobx: 5.13.0

**Comment by urugator** [comment]
I think the problem is here:
https://github.com/mobxjs/mobx/blob/master/src/api/become-observed.ts#L50-L51

It should go like:
```javascript 
const atom: IObservable =
   typeof arg3 === "function" ? getAtom(thing, arg2) : (getAtom(thing) as any)
const cb = typeof arg3 === "function" ? arg3 : arg2
```
PR welcome

**Comment by gespiton** [comment]
ok, I'll try

**Comment by gespiton** [comment]
pr is made: https://github.com/mobxjs/mobx/pull/2073

**Comment by FredyC** [comment]
PRs merged for both versions #2073 #2075
